dev-tools/check_project.sh

```diff
@@ -32,11 +32,36 @@ if [[ ${EUID} -eq 0 ]]; then
   echo -e "\\n${RED}""Running this script as root is not supported""${NC}\\n"
 fi
 
+declare -A lastCheckedCache
```

Comment:
All variable names should be Capitalized to be consistent
```suggestion
declare -A LAST_CHECKED_CACHE
```